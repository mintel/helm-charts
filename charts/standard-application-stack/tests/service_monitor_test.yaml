suite: Test ServiceMonitors
templates:
  - deployment.yaml
  - service.yaml
  - service-monitor.yaml
release:
  namespace: test-namespace
tests:
  - it: Check name, namespace and labels
    template: service-monitor.yaml
    set:
      global.name: test-app
      global.clusterEnv: prod
      service.enabled: true
      metrics.enabled: true
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: ServiceMonitor
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: test-app
      - equal:
          path: metadata.labels["app.kubernetes.io/name"]
          value: test-app
      - equal:
          path: metadata.labels["name"]
          value: test-app
  - it: Check default targetLabels
    template: service-monitor.yaml
    set:
      global.name: test-app
      global.clusterEnv: prod
      service.enabled: true
      metrics.enabled: true
    asserts:
      - equal:
          path: spec.targetLabels
          value:
            - app.mintel.com/owner
            - app.mintel.com/ignore_alerts
      - equal:
          path: spec.podTargetLabels
          value:
            - app.kubernetes.io/name
            - app.kubernetes.io/part-of
            - app.kubernetes.io/component
            - app.kubernetes.io/instance
            - app.mintel.com/owner
  - it: Check default endpoints
    template: service-monitor.yaml
    set:
      global.name: test-app
      global.clusterEnv: prod
      service.enabled: true
      metrics.enabled: true
    asserts:
      - isKind:
          of: ServiceMonitor
      - equal:
          path: spec.endpoints
          value:
            - interval: 30s
              path: /metrics
              port: main-http
              relabelings:
                - separator: /
                  sourceLabels:
                    - __meta_kubernetes_namespace
                    - __meta_kubernetes_pod_label_app_kubernetes_io_name
                  targetLabel: job
              scheme: http
              scrapeTimeout: 10s
              tlsConfig:
                insecureSkipVerify: true
  - it: Check metric overrides
    template: service-monitor.yaml
    set:
      global.name: test-app
      global.clusterEnv: prod
      service.enabled: true
      metrics.enabled: true
      metrics.path: /updated-metrics-path
      metrics.port: another-named-port
      metrics.scheme: https
      metrics.timeout: 20s
    asserts:
      - isKind:
          of: ServiceMonitor
      - equal:
          path: spec.endpoints
          value:
            - interval: 30s
              path: /updated-metrics-path
              port: main-another-named-port
              relabelings:
                - separator: /
                  sourceLabels:
                    - __meta_kubernetes_namespace
                    - __meta_kubernetes_pod_label_app_kubernetes_io_name
                  targetLabel: job
              scheme: https
              scrapeTimeout: 20s
              tlsConfig:
                insecureSkipVerify: true
  - it: Check basic-auth
    template: service-monitor.yaml
    set:
      global.name: test-app
      global.clusterEnv: prod
      service.enabled: true
      metrics.enabled: true
      metrics.basicAuth.enabled: true
      metrics.basicAuth.usernameKey: user
      metrics.basicAuth.passwordKey: passwd
    asserts:
      - isKind:
          of: ServiceMonitor
      - equal:
          path: spec.endpoints
          value:
            - interval: 30s
              path: /metrics
              port: main-http
              basicAuth:
                username:
                  key: user
                  name: test-app-app
                password:
                  key: passwd
                  name: test-app-app
              relabelings:
                - separator: /
                  sourceLabels:
                    - __meta_kubernetes_namespace
                    - __meta_kubernetes_pod_label_app_kubernetes_io_name
                  targetLabel: job
              scheme: http
              scrapeTimeout: 10s
              tlsConfig:
                insecureSkipVerify: true
  - it: Check additional ServiceMonitor resources
    template: service-monitor.yaml
    set:
      global.name: test-app
      global.clusterEnv: prod
      service.enabled: true
      metrics.enabled: true
      metrics.additionalMonitors:
        - name: metrics-2
          path: /metrics-2
        - name: metrics-3
          path: /metrics-3
          interval: 20s
          scheme: https
    asserts:
      - hasDocuments:
          count: 3
      - isKind:
          of: ServiceMonitor
        documentIndex: 0
      - equal:
          path: metadata.name
          value: test-app
        documentIndex: 0
      - isKind:
          of: ServiceMonitor
        documentIndex: 1
      - equal:
          path: metadata.name
          value: test-app-metrics-2
        documentIndex: 1
      - isKind:
          of: ServiceMonitor
        documentIndex: 2
      - equal:
          path: metadata.name
          value: test-app-metrics-3
        documentIndex: 2
      - equal:
          path: spec.endpoints
          value:
            - interval: 30s
              path: /metrics-2
              port: main-metrics-2
              relabelings:
                - separator: /
                  sourceLabels:
                    - __meta_kubernetes_namespace
                    - __meta_kubernetes_pod_label_app_kubernetes_io_name
                  targetLabel: job
              scheme: http
              scrapeTimeout: 10s
              tlsConfig:
                insecureSkipVerify: true
        documentIndex: 1
      - equal:
          path: spec.endpoints
          value:
            - interval: 20s
              path: /metrics-3
              port: main-metrics-3
              relabelings:
                - separator: /
                  sourceLabels:
                    - __meta_kubernetes_namespace
                    - __meta_kubernetes_pod_label_app_kubernetes_io_name
                  targetLabel: job
              scheme: https
              scrapeTimeout: 10s
              tlsConfig:
                insecureSkipVerify: true
        documentIndex: 2
  - it: Check Service defaults
    template: service.yaml
    set:
      global.name: test-app
      global.clusterEnv: prod
      service.enabled: true
      metrics.enabled: true
    asserts:
      - isKind:
          of: Service
      - equal:
          path: spec.ports
          value:
            - name: main-http
              port: 8000
              targetPort: 8000
  - it: Check combined template defaults
    set:
      global.name: test-app
      global.clusterEnv: prod
      service.enabled: true
      metrics.enabled: true
    asserts:
      - isKind:
          of: Service
        template: service.yaml
      - equal:
          path: spec.ports
          value:
            - name: main-http
              port: 8000
              targetPort: 8000
        template: service.yaml
      - isKind:
          of: ServiceMonitor
        template: service-monitor.yaml
      - equal:
          path: spec.endpoints
          value:
            - interval: 30s
              path: /metrics
              port: main-http
              relabelings:
                - separator: /
                  sourceLabels:
                    - __meta_kubernetes_namespace
                    - __meta_kubernetes_pod_label_app_kubernetes_io_name
                  targetLabel: job
              scheme: http
              scrapeTimeout: 10s
              tlsConfig:
                insecureSkipVerify: true
        template: service-monitor.yaml
      - isKind:
          of: Deployment
        template: deployment.yaml
      - equal:
          path: spec.template.spec.containers[0].ports
          value:
            - containerPort: 8000
              name: http
        template: deployment.yaml
  - it: Check combined template with extra ports and monitors
    set:
      global.name: test-app
      global.clusterEnv: prod
      service.enabled: true
      metrics.enabled: true
      metrics.additionalMonitors:
        - name: metrics-2
          path: /metrics-2
        - name: metrics-3
          path: /metrics-3
          interval: 20s
          scheme: https
      extraPorts:
        - name: metrics-2
          containerPort: 9002
        - name: metrics-3
          containerPort: 9003
    asserts:
      - isKind:
          of: Service
        template: service.yaml
      - equal:
          path: spec.ports
          value:
            - name: main-http
              port: 8000
              targetPort: 8000
            - name: main-metrics-2
              port: 9002
              targetPort: 9002
            - name: main-metrics-3
              port: 9003
              targetPort: 9003
        template: service.yaml
      - isKind:
          of: ServiceMonitor
        template: service-monitor.yaml
      - equal:
          path: spec.endpoints
          value:
            - interval: 30s
              path: /metrics
              port: main-http
              relabelings:
                - separator: /
                  sourceLabels:
                    - __meta_kubernetes_namespace
                    - __meta_kubernetes_pod_label_app_kubernetes_io_name
                  targetLabel: job
              scheme: http
              scrapeTimeout: 10s
              tlsConfig:
                insecureSkipVerify: true
        documentIndex: 0
        template: service-monitor.yaml
      - equal:
          path: spec.endpoints
          value:
            - interval: 30s
              path: /metrics-2
              port: main-metrics-2
              relabelings:
                - separator: /
                  sourceLabels:
                    - __meta_kubernetes_namespace
                    - __meta_kubernetes_pod_label_app_kubernetes_io_name
                  targetLabel: job
              scheme: http
              scrapeTimeout: 10s
              tlsConfig:
                insecureSkipVerify: true
        documentIndex: 1
        template: service-monitor.yaml
      - equal:
          path: spec.endpoints
          value:
            - interval: 20s
              path: /metrics-3
              port: main-metrics-3
              relabelings:
                - separator: /
                  sourceLabels:
                    - __meta_kubernetes_namespace
                    - __meta_kubernetes_pod_label_app_kubernetes_io_name
                  targetLabel: job
              scheme: https
              scrapeTimeout: 10s
              tlsConfig:
                insecureSkipVerify: true
        documentIndex: 2
        template: service-monitor.yaml
      - isKind:
          of: Deployment
        template: deployment.yaml
      - equal:
          path: spec.template.spec.containers[0].ports
          value:
            - containerPort: 8000
              name: http
            - containerPort: 9002
              name: metrics-2
            - containerPort: 9003
              name: metrics-3
        template: deployment.yaml
