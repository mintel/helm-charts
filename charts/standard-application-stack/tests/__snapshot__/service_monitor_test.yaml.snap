Check Service defaults:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations: null
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      ports:
        - name: main-http
          port: 8000
          targetPort: 8000
      selector:
        app.kubernetes.io/component: app
        app.kubernetes.io/name: test-app
      type: ClusterIP
Check additional ServiceMonitor resources:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        k8s-app: test-app
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      endpoints:
        - interval: 30s
          path: /metrics
          port: main-http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      targetLabels:
        - app.mintel.com/owner
        - app.mintel.com/ignore_alerts
  2: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: metrics-2
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app-metrics-2
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        k8s-app: test-app-metrics-2
        name: test-app-metrics-2
      name: test-app-metrics-2
      namespace: test-namespace
    spec:
      endpoints:
        - interval: 30s
          path: /metrics-2
          port: main-metrics-2
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: metrics-2
          app.kubernetes.io/name: test-app-metrics-2
      targetLabels:
        - app.mintel.com/owner
        - app.mintel.com/ignore_alerts
  3: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: metrics-3
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app-metrics-3
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        k8s-app: test-app-metrics-3
        name: test-app-metrics-3
      name: test-app-metrics-3
      namespace: test-namespace
    spec:
      endpoints:
        - interval: 20s
          path: /metrics-3
          port: main-metrics-3
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: https
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: metrics-3
          app.kubernetes.io/name: test-app-metrics-3
      targetLabels:
        - app.mintel.com/owner
        - app.mintel.com/ignore_alerts
Check basic-auth:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        k8s-app: test-app
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      endpoints:
        - basicAuth:
            password:
              key: passwd
              name: test-app-app
            username:
              key: user
              name: test-app-app
          interval: 30s
          path: /metrics
          port: main-http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      targetLabels:
        - app.mintel.com/owner
        - app.mintel.com/ignore_alerts
Check combined template defaults:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        helm.sh/chart: standard-application-stack-3.61.5
        secret.reloader.stakater.com/reload: test-app-app
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          annotations:
            instrumentation.opentelemetry.io/inject-sdk: test-namespace/test-app
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            app.mintel.com/env: prod
            app.mintel.com/region: ${CLUSTER_REGION}
            name: test-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: prod
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: test-app-app
              image: registry.gitlab.com/test:auto-replaced
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            runAsUser: 1000
          serviceAccountName: test-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: test-app
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: test-app
              maxSkew: 1
              topologyKey: kubernetes.io/hostname
              whenUnsatisfiable: DoNotSchedule
          volumes: null
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations: null
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      ports:
        - name: main-http
          port: 8000
          targetPort: 8000
      selector:
        app.kubernetes.io/component: app
        app.kubernetes.io/name: test-app
      type: ClusterIP
  3: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        k8s-app: test-app
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      endpoints:
        - interval: 30s
          path: /metrics
          port: main-http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      targetLabels:
        - app.mintel.com/owner
        - app.mintel.com/ignore_alerts
Check combined template with extra ports and monitors:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        helm.sh/chart: standard-application-stack-3.61.5
        secret.reloader.stakater.com/reload: test-app-app
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          annotations:
            instrumentation.opentelemetry.io/inject-sdk: test-namespace/test-app
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            app.mintel.com/env: prod
            app.mintel.com/region: ${CLUSTER_REGION}
            name: test-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: prod
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: test-app-app
              image: registry.gitlab.com/test:auto-replaced
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
                - containerPort: 9002
                  name: metrics-2
                - containerPort: 9003
                  name: metrics-3
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            runAsUser: 1000
          serviceAccountName: test-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: test-app
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: test-app
              maxSkew: 1
              topologyKey: kubernetes.io/hostname
              whenUnsatisfiable: DoNotSchedule
          volumes: null
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations: null
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      ports:
        - name: main-http
          port: 8000
          targetPort: 8000
        - name: main-metrics-2
          port: 9002
          targetPort: 9002
        - name: main-metrics-3
          port: 9003
          targetPort: 9003
      selector:
        app.kubernetes.io/component: app
        app.kubernetes.io/name: test-app
      type: ClusterIP
  3: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        k8s-app: test-app
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      endpoints:
        - interval: 30s
          path: /metrics
          port: main-http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      targetLabels:
        - app.mintel.com/owner
        - app.mintel.com/ignore_alerts
  4: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: metrics-2
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app-metrics-2
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        k8s-app: test-app-metrics-2
        name: test-app-metrics-2
      name: test-app-metrics-2
      namespace: test-namespace
    spec:
      endpoints:
        - interval: 30s
          path: /metrics-2
          port: main-metrics-2
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: metrics-2
          app.kubernetes.io/name: test-app-metrics-2
      targetLabels:
        - app.mintel.com/owner
        - app.mintel.com/ignore_alerts
  5: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: metrics-3
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app-metrics-3
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        k8s-app: test-app-metrics-3
        name: test-app-metrics-3
      name: test-app-metrics-3
      namespace: test-namespace
    spec:
      endpoints:
        - interval: 20s
          path: /metrics-3
          port: main-metrics-3
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: https
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: metrics-3
          app.kubernetes.io/name: test-app-metrics-3
      targetLabels:
        - app.mintel.com/owner
        - app.mintel.com/ignore_alerts
Check default endpoints:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        k8s-app: test-app
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      endpoints:
        - interval: 30s
          path: /metrics
          port: main-http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      targetLabels:
        - app.mintel.com/owner
        - app.mintel.com/ignore_alerts
Check default targetLabels:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        k8s-app: test-app
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      endpoints:
        - interval: 30s
          path: /metrics
          port: main-http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      targetLabels:
        - app.mintel.com/owner
        - app.mintel.com/ignore_alerts
Check metric overrides:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        k8s-app: test-app
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      endpoints:
        - interval: 30s
          path: /updated-metrics-path
          port: main-another-named-port
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: https
          scrapeTimeout: 20s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      targetLabels:
        - app.mintel.com/owner
        - app.mintel.com/ignore_alerts
Check name, namespace and labels:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        k8s-app: test-app
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      endpoints:
        - interval: 30s
          path: /metrics
          port: main-http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      targetLabels:
        - app.mintel.com/owner
        - app.mintel.com/ignore_alerts
